ngraph/cmake/patches/onnx_patch.diff

```diff
@@ -0,0 +1,30 @@
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index 5254f7ee..e7a0ce2b 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -380,7 +380,8 @@ if(CMAKE_SYSTEM_NAME STREQUAL "AIX")
+   # So, create a object library
+   add_library(onnx OBJECT ${ONNX_SRCS})
+ else()
+-  add_library(onnx ${ONNX_SRCS})
++  # onnx target doesn't export symbols
++  add_library(onnx STATIC ${ONNX_SRCS})
```

Comment:
Can we use [BUILD_SHARED_LIBS=OFF](https://cmake.org/cmake/help/latest/variable/BUILD_SHARED_LIBS.html#variable:BUILD_SHARED_LIBS) in order to avoid custom patches?

Reply:
As I wrote in description from onnx cmake we use two targets: `onnx` and `onnx_proto`.
We need only `onnx_proto` to be shared (especially because `onnx` doesn't expose symbols correctly).
As I know there is no possibility to configure FetchContent to select which target will be shared.
In [issue](https://github.com/onnx/onnx/issues/3319) I propose to build `onnx` as static explicitly.

Reply:
We understand that we need it to be shared. Just can we don't use patch and use external option to achieve the same result?

Reply:
Result of current patch is that:
- `onnx` target is static, `onnx_proto` target is shared

Result of `BUILD_SHARED_LIBS=OF` is:
- both `onnx` and `onnx_proto` targets are static

onnx library cmake doesn't provide any option to control it, but we propose it as a part of [issue](https://github.com/onnx/onnx/issues/3319), but with explicitly indicate onnx as static we achieve the same.

In current moment (before introduction changes to onnx repo) I am not aware of any method which allows to make only `onnx_proto` shared (without patch)